lib/Dialect/Torch/Transforms/Passes.cpp

```diff
@@ -111,9 +102,18 @@ void mlir::torch::Torch::createTorchOnnxToTorchBackendPipeline(
         Torch::createDecomposeComplexOpsPass(options.backendLegalOps));
     pm.addNestedPass<func::FuncOp>(createCanonicalizerPass());
   }
-  createOnnxShapeRefinementPipeline(pm, options);
+  // TODO: Move the below two passes i.e., ScalarizeShapes and
+  // TorchShapeRefinementPipeline out of here and create an onnx shape
+  // refinement pipeline which runs iteratively over the IR.
+  // This pass scalarizes the tensor shape computations.
```

Comment:
Now that I think about it, at least until we perform the shape refinement/scalarization iteratively, let's include one pass of shape refinement before scalarize shapes, too.

Shape scalarization leans pretty heavily on folding `aten.where.self` operations, and sometimes this is only possible when a `size.int` op is foldable to a constant.